department-of-veterans-affairs/vets-api
# SC 4142 status updater job

## Summary

- *This work is behind a feature toggle (flipper): YES: decision_review_track_4142_submissions*
- The job will continue to run on its regular schedule, but this new functionality won't run until the Flipper is enabled
- polls and updates status of 4142 forms submitted during Supplemental Claim to detect silent failures
- logs metrics
- Decision Reviews, yes


## Related issue(s)

department-of-veterans-affairs/va.gov-team/issues/95084

## Testing done

- We were never checking on the status of submitted 4142s
- Run the job in staging, see status change for relevant records
- Tests for job in both cases. When disabled the job is a no-op
  - Enable for a percentage of users, verify it can run regularly without error, enable for all

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
No user-facing changes. Expands the SavedClaimScStatusUpdaterJob to check status of 4142s. This only applies for 4142s submitted as part of submitting a Supplemental Claim. Doesn't include a chron entry, will test job in production before adding a scheduled entry.

## Acceptance criteria

- [ ]  Feature/bug has a monitor built into Datadog (if applicable) - need to build monitor once this goes live
